### PR TITLE
Pad the LZW code table if less than minimum size, alt fix

### PR DESCRIPTION
**Pull request by rendaw**
Sorry for the duplicate PRs, I wasn't sure which approach was better and rather than wait for a response I figured I'd try this new approach.

Continuing from #14 , it looks like it's enough to just fix numColors with a minimum.  I also added a docstring indicating that the input value must be a power of 2.

### CODE REVIEW

**Comment by dlubarov** [comment]
LGTM; @JakeWharton can you merge it?

File: src/main/java/com/squareup/gifencoder/LzwEncoder.java [hunk 1]
```diff
@@ -44,8 +44,11 @@
   private int codeSize;
   private List<Integer> indexBuffer = new ArrayList<>();
 
+  /**
+   * @param numColors Padded number of colors, must be a power of 2
```
Comment (JakeWharton) [inline src/main/java/com/squareup/gifencoder/LzwEncoder.java]: Should we validate this and throw IllegalArgumentException otherwise?

**Comment by JakeWharton** [comment]
Rebased on master to fix CI and squash into a single commit.

File: src/main/java/com/squareup/gifencoder/LzwEncoder.java (hunk 1 above)
Comment (dlubarov) [reply src/main/java/com/squareup/gifencoder/LzwEncoder.java]: Will do in a a followup PR.

**Pull request merged**